### PR TITLE
Support Bazel 6.0.0

### PR DESCRIPTION
**Pull request by tpasternak**
1. Main Repository Target Label Prefixes.
In Bazel 5 and older, the target labels were strigified to "//"-prefixed names. Since
6.0.0, they are stringified to "@//"-prefixed names. Unfortunately, this does not apply
to data coming from BEP. Targets references in BEP events payload are "//"-prefixed.
For this reason we need to add "@" manually after retrieving data from BEP.

2. aspects.bzl: Retrieve runtime toolchain in the new way (Bazel 6.0.0)

3. cpp-project: Old GTest version is incompatible with Bazel 6.0.0, the
new one required cpp version bump (--std=c++17)

4. e2e tests are now adapted to run across multiple Bazel versions (currently 5 and 6)

5. JdkResolver: JDK version was inferred from source version, which is not always true. That's
why it is safer just to pass null.

### CODE REVIEW

**Review by lukaszwawrzyk — APPROVED** [review]

File: bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+) {
+  val major: Int by lazy {
+    """(?<=release )\d+(?=[0-9.]*)""".toRegex().find(versionString)?.value?.toInt()!!
+  }
+
+  fun mainRepositoryReferencePrefix() = when(major) {
+    in 0..3 -> throw RuntimeException("Unsupported Bazel version, use Bazel 4 or newer")
+    in 4..5 -> "//"
+    else -> "@//"
+  }
```
Comment (abrams27) [inline bazelrunner/src/main/java/org/jetbrains/bsp/bazel/bazelrunner/BazelInfo.kt:23]: can we have tests for it? including this issue as well: https://github.com/JetBrains/bazel-bsp/issues/92

File: e2e/src/main/java/org/jetbrains/bsp/bazel/base/BazelBspTestBaseScenario.kt [hunk 2]
```diff
@@ -9,6 +9,9 @@ import java.nio.file.Paths
 import kotlin.system.exitProcess
 
 abstract class BazelBspTestBaseScenario {
+    val targetPrefix = Runtime.getRuntime().exec("bazel version").inputStream.bufferedReader().readText()
+            .let {"""(?<=Build label: )\d+(?=[0-9.]+)""".toRegex().find(it)!!.value.toInt() }
```
Comment (abrams27) [inline e2e/src/main/java/org/jetbrains/bsp/bazel/base/BazelBspTestBaseScenario.kt:13]: do we need triple " here?

File: server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/java/JdkResolver.kt [hunk 3]
```diff
@@ -59,14 +59,9 @@ class JdkResolver(
           null
     val javaHome = javaHomeFile?.let { bazelPathsResolver.resolveUri(it) }
 
-    val version =
-        if (targetInfo.hasJavaToolchainInfo())
-          targetInfo.javaToolchainInfo.sourceVersion
-        else
-          null
-
+    val version = null
```
Comment (abrams27) [inline server/src/main/java/org/jetbrains/bsp/bazel/server/sync/languages/java/JdkResolver.kt:62]: now when the version is null the mechanizm in `JdkCandidate` for the `version` field should be updated (in general we can remove `version` from `JdkCandidateData`)

**Review by abrams27 — COMMENTED** [review]
i feel that we can do it without bazel version - the bazel version is used only in 3 places:
- `isWorkspaceTarget` - we can just check does the id starts with `//` or `@//`
- similar for extracting the relative path

but then we could do it without passing `BazelInfo` around, what d u think about it?

**Review by abrams27 — APPROVED** [review]

**Pull request merged**